### PR TITLE
Fix backup file path substitution

### LINKED ISSUE #10505: Backup file placeholders for time not working as expected
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )
Some Date substitution in to placeholders results in unexpected results

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. In preferences in the backup destination field I added custom path 

`./{TIME:yy}/{DB_FILENAME }_{TIME:yyyyMMdd_HHmmss}.old.kdbx`

expecting it to create  a backup file such as
`./2024/CJDatabase_20240326_121111.old.kdbx`

but instead it creates

`./Year_2024}/CJ-DpmGMTpmbpm10e_{TI3E:20240326_121010.old.kdbx`

2. if i use a simple time without formatting  such as
`./{TIME}/{DB_FILENAME}_{TIME}.old.kdbx`

if has the expected results of creating a file in a folder

`./26_03_2024_12-32-40/CJ-Database_26_03_2024_12-32-40.old.kdbx`

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
  - I expected {TIME:yyyy} to be replaced by 2024
  - I expected {DB_FILENAME} to dvbe replaced by CJ-Database in filenames that also have custom time formats

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
using

`./{TIME:yy}/{DB_FILENAME }_{TIME:yyyyMMdd_HHmmss}.old.kdbx`

I expected it to create  a backup file such as
`./2024/CJDatabase_20240326_121111.old.kdbx`

but instead it creates

`./Year_2024}/CJ-DpmGMTpmbpm10e_{TI3E:20240326_121010.old.kdbx`

## Context
[NOTE]: # ( Give us any additional information you may have. )

On further testing it appears to be if I include two {TIME} placeholders in the backup path, and either or both have  custom formats then the whole file name gets corrupted

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.7.7
Revision: 68e2dd8

Qt 5.15.11
Debugging mode is disabled.

Operating system: macOS 14.4
CPU architecture: arm64
Kernel: darwin 23.4.0

Enabled extensions:
- Auto-Type
- Browser Integration
- Passkeys
- SSH Agent
- KeeShare
- YubiKey
- Quick Unlock

Cryptographic libraries:
- Botan 3.1.1

[NOTE]: # ( Pick choices based on your environment )
Operating System: macOS
Desktop Env: macos

### PR DESCRIPTION
**Pull request by c4rlo**
Previously, in a pattern like "`{TIME:yy} {TIME}`", `Tools::substituteBackupFilePath()` would greedily use the entire string "`yy} {TIME`" as the format specifier for the first TIME template, instead of just "`yy`". Fix this, by adjusting the regular expression.

This ends up changing the behaviour of a weird corner case that is covered in the tests, so change the test. I don't think anyone cares about that case, and I think the current behaviour is better there.

Fixes #10505 (proved by adding a test case very similar to what was reported there).

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change that fixes an issue)

### CODE REVIEW

File: src/core/Tools.cpp [hunk 1]
```diff
@@ -451,26 +451,25 @@ namespace Tools
...
-        // Replace backup pattern
-        QFileInfo dbFileInfo(databasePath);
-        QString baseName = dbFileInfo.completeBaseName();
+        const QString baseName = QFileInfo{databasePath}.completeBaseName();
 
-        pattern.replace(QString("{DB_FILENAME}"), baseName);
+        pattern.replace("{DB_FILENAME}", baseName);
 
-        auto re = QRegularExpression(R"(\{TIME(?::([^\\]*))?\})");
+        const QDateTime now = Clock::currentDateTime();
```
Comment (c4rlo) [inline src/core/Tools.cpp:462]: Unrelated to the issue I'm fixing, but I pulled this out of the loop to ensure a consistent timestamp is used for all substitutions.

**Comment by droidmonkey** [comment]
Agree the corner case isn't a valid substitution syntax anyway

**Review by droidmonkey — APPROVED** [review]
